src/broker/broker.c

```diff
@@ -921,6 +924,41 @@ static int init_local_uri_attr (struct overlay *ov, attr_t *attrs)
     return 0;
 }
 
+static int init_critical_ranks_attr (struct overlay *ov, attr_t *attrs)
+{
+    int rc = -1;
+    const char *val;
+    char *ranks = NULL;
+    struct idset *critical_ranks = NULL;
+
+    if (attr_get (attrs, "broker.critical-ranks", &val, NULL) < 0) {
+        if (!(critical_ranks = overlay_get_critical_ranks (ov))
+            || !(ranks = idset_encode (critical_ranks, IDSET_FLAG_RANGE))) {
+            log_err ("unable to calculate critical-ranks attribute");
+            goto out;
+        }
+        if (attr_add (attrs,
+                      "broker.critical-ranks",
+                      ranks,
+                      FLUX_ATTRFLAG_IMMUTABLE) < 0) {
+            log_err ("attr_add critical_ranks");
+            goto out;
+        }
+    }
+    else {
+        if (!(critical_ranks = idset_decode (val))
+            || idset_last (critical_ranks) >= overlay_get_size (ov)) {
+            log_msg ("invalid value for broker.critical-ranks='%s'", val);
+            goto out;
+        }
+    }
```

Comment:
Ugh so sorry about this broker attr api  :frowning_person: 

You'll want to set the IMMUTABLE flag on the attribute if it came from the command line so that people won't think it can be set on the fly.
```c
if (attr_set_flags (attrs, "broker.critical_ranks", FLUX_ATTRFLAG_IMMUTABLE) < 0)
    goto out;
```